### ISSUE TITLE
[MNT]: Failing CI on OSX

### ISSUE THREAD

**Issue by timhoffm** [matplotlib/matplotlib#23094]
### Summary

Our current OSX pipelines fail: I've collected the information from one of the runs

The stripped down version is:
```
2022-05-21T22:18:40.8636500Z __________________________________ test_blit ___________________________________
2022-05-21T22:18:40.8800310Z E           RuntimeError: tk.h version (8.6) doesn't match libtk.a version (8.5)
2022-05-21T22:18:40.8800760Z 
2022-05-21T22:18:40.8801280Z lib/matplotlib/tests/test_backend_tk.py:54: Failed

2022-05-21T22:18:40.8801980Z ____________________ test_figuremanager_cleans_own_mainloop ____________________
2022-05-21T22:18:40.9014710Z E           RuntimeError: tk.h version (8.6) doesn't match libtk.a version (8.5)
2022-05-21T22:18:40.9015170Z 
2022-05-21T22:18:40.9015690Z lib/matplotlib/tests/test_backend_tk.py:54: Failed

2022-05-21T22:18:40.9016410Z ___________________________ test_missing_back_button ___________________________
2022-05-21T22:18:40.9185730Z E           RuntimeError: tk.h version (8.6) doesn't match libtk.a version (8.5)
2022-05-21T22:18:40.9186350Z 
2022-05-21T22:18:40.9186900Z lib/matplotlib/tests/test_backend_tk.py:54: Failed

2022-05-21T22:18:40.9187580Z __________________ test_figuremanager_preserves_host_mainloop __________________
2022-05-21T22:18:40.9358480Z E           RuntimeError: tk.h version (8.6) doesn't match libtk.a version (8.5)
2022-05-21T22:18:40.9359080Z 
2022-05-21T22:18:40.9359580Z lib/matplotlib/tests/test_backend_tk.py:54: Failed

2022-05-21T22:18:40.9360270Z ________________________________ test_embedding ________________________________
2022-05-21T22:18:40.9526200Z E           RuntimeError: tk.h version (8.6) doesn't match libtk.a version (8.5)
2022-05-21T22:18:40.9526560Z 
2022-05-21T22:18:40.9527090Z lib/matplotlib/tests/test_backend_tk.py:54: Failed

2022-05-21T22:18:40.9528520Z __________ test_interactive_backend[toolbar2-{'MPLBACKEND': 'tkagg'}] __________
022-05-21T22:18:40.9773130Z E               subprocess.CalledProcessError: Command '['/Users/runner/hostedtoolcache/Python/3.8.12/x64/bin/python', '-c', '\nfrom matplotlib.tests.test_backends_interactive import _test_interactive_impl\n_test_interactive_impl()\n', '{"toolbar": "toolbar2"}']' returned non-zero exit status 1.
2022-05-21T22:18:40.9774220Z 
2022-05-21T22:18:40.9774850Z ../../../hostedtoolcache/Python/3.8.12/x64/lib/python3.8/subprocess.py:516: CalledProcessError

2022-05-21T22:18:40.9776050Z ________________ test_blitting_events[{'MPLBACKEND': 'tkagg'}] _________________
2022-05-21T22:18:40.9894890Z E               subprocess.CalledProcessError: Command '['/Users/runner/hostedtoolcache/Python/3.8.12/x64/bin/python', '-c', '\nfrom matplotlib.tests.test_backends_interactive import _test_number_of_draws_script\n_test_number_of_draws_script()\n']' returned non-zero exit status 1.
2022-05-21T22:18:40.9895970Z 
2022-05-21T22:18:40.9896600Z ../../../hostedtoolcache/Python/3.8.12/x64/lib/python3.8/subprocess.py:516: CalledProcessError

2022-05-21T22:18:40.9897820Z __________ test_figure_leak_20490[time_mem0-{'MPLBACKEND': 'tkagg'}] ___________
2022-05-21T22:18:40.9970990Z E               subprocess.CalledProcessError: Command '['/Users/runner/hostedtoolcache/Python/3.8.12/x64/bin/python', '-c', '\nfrom matplotlib.tests.test_backends_interactive import _test_figure_leak\n_test_figure_leak()\n', '0.0']' returned non-zero exit status 1.
2022-05-21T22:18:40.9971810Z 
2022-05-21T22:18:40.9972440Z ../../../hostedtoolcache/Python/3.8.12/x64/lib/python3.8/subprocess.py:516: CalledProcessError

2022-05-21T22:18:40.9973650Z __________ test_figure_leak_20490[time_mem1-{'MPLBACKEND': 'tkagg'}] ___________
2022-05-21T22:18:41.0242530Z E               subprocess.CalledProcessError: Command '['/Users/runner/hostedtoolcache/Python/3.8.12/x64/bin/python', '-c', '\nfrom matplotlib.tests.test_backends_interactive import _test_figure_leak\n_test_figure_leak()\n', '0.1']' returned non-zero exit status 1.
2022-05-21T22:18:41.0243350Z 
2022-05-21T22:18:41.0243970Z ../../../hostedtoolcache/Python/3.8.12/x64/lib/python3.8/subprocess.py:516: CalledProcessError

2022-05-21T22:18:41.0245150Z ________ test_interactive_backend[toolmanager-{'MPLBACKEND': 'tkagg'}] _________
2022-05-21T22:18:41.1102640Z E               subprocess.CalledProcessError: Command '['/Users/runner/hostedtoolcache/Python/3.8.12/x64/bin/python', '-c', '\nfrom matplotlib.tests.test_backends_interactive import _test_interactive_impl\n_test_interactive_impl()\n', '{"toolbar": "toolmanager"}']' returned non-zero exit status 1.
2022-05-21T22:18:41.1103670Z 
2022-05-21T22:18:41.1104290Z ../../../hostedtoolcache/Python/3.8.12/x64/lib/python3.8/subprocess.py:516: CalledProcessError

2022-05-21T22:18:41.1105500Z ___________ test_interactive_thread_safety[{'MPLBACKEND': 'tkagg'}] ____________
2022-05-21T22:18:41.1167240Z E               subprocess.CalledProcessError: Command '['/Users/runner/hostedtoolcache/Python/3.8.12/x64/bin/python', '-c', '\nfrom matplotlib.tests.test_backends_interactive import _test_thread_impl\n_test_thread_impl()\n']' returned non-zero exit status 1.
2022-05-21T22:18:41.1167960Z 
2022-05-21T22:18:41.1168480Z ../../../hostedtoolcache/Python/3.8.12/x64/lib/python3.8/subprocess.py:516: 
```

Observations:

- All of the "RuntimeError: tk.h version (8.6) doesn't match libtk.a version (8.5)" happens in tests that are decorated with `@_isolated_tk_test()`. We have a couple more uses of the decorator, which also have `@pytest.mark.backend('TkAgg', skip_on_importerror=True)` and thus don't fail.

**Comment by QuLogic** [comment]
See upstream issue https://github.com/actions/setup-python/issues/402

**Comment by QuLogic** [comment]
This was "fixed" by #23095, and then really fixed by upstream, which was confirmed by the revert in #23355.
